src/lib/badge/badge.ts

```diff
@@ -0,0 +1,180 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, Input, ElementRef, Inject, Optional, NgZone} from '@angular/core';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {ThemePalette} from '@angular/material/core';
+import {AriaDescriber} from '@angular/cdk/a11y';
+import {DOCUMENT} from '@angular/common';
+
+let nextId = 0;
+
+export type PositionTypes =
```

Comment:
Not really sure about this, but I guess that all (*public*) interfaces/types must be prefixed with `Mat#{ComponentName/DirectiveName}`. The same applies for `BadgeSize`.

Reply:
Yeah, this one should be `MatBadgePosition`. It will also fit on one line.